localstack/services/firehose/provider.py

```diff
@@ -0,0 +1,610 @@
+
+# global sequence number counter for Firehose records (these are very large long values in AWS)
+SEQUENCE_NUMBER = 49546986683135544286507457936321625675700192471156785154
+SEQUENCE_NUMBER_MUTEX = threading.RLock()
+
+
+def next_sequence_number() -> int:
+    """Increase and return the next global sequence number."""
+    global SEQUENCE_NUMBER
+    with SEQUENCE_NUMBER_MUTEX:
+        SEQUENCE_NUMBER += 1
+        return SEQUENCE_NUMBER
+
+
+class FirehoseBackend(RegionBackend):
+    # maps delivery stream names to DeliveryStreamDescription
+    delivery_streams: Dict[str, DeliveryStreamDescription]
+    # static tagging service instance
+    tags = TaggingService()
+
+    def __init__(self):
+        self.delivery_streams = {}
+
+
+def _get_description_or_raise_not_found(
+    context, delivery_stream_name: str
+) -> DeliveryStreamDescription:
+    region = FirehoseBackend.get()
+    delivery_stream_description = region.delivery_streams.get(delivery_stream_name)
+    if not delivery_stream_description:
+        raise ResourceNotFoundException(
+            f"Firehose {delivery_stream_name} under account {context.account_id} " f"not found."
+        )
+    return delivery_stream_description
+
+
+class FirehoseProvider(FirehoseApi):
+    def create_delivery_stream(
+        self,
+        context: RequestContext,
+        delivery_stream_name: DeliveryStreamName,
+        delivery_stream_type: DeliveryStreamType = DeliveryStreamType.DirectPut,
+        kinesis_stream_source_configuration: KinesisStreamSourceConfiguration = None,
+        delivery_stream_encryption_configuration_input: DeliveryStreamEncryptionConfigurationInput = None,
+        s3_destination_configuration: S3DestinationConfiguration = None,
+        extended_s3_destination_configuration: ExtendedS3DestinationConfiguration = None,
+        redshift_destination_configuration: RedshiftDestinationConfiguration = None,
+        elasticsearch_destination_configuration: ElasticsearchDestinationConfiguration = None,
+        amazonopensearchservice_destination_configuration: AmazonopensearchserviceDestinationConfiguration = None,
+        splunk_destination_configuration: SplunkDestinationConfiguration = None,
+        http_endpoint_destination_configuration: HttpEndpointDestinationConfiguration = None,
+        tags: TagDeliveryStreamInputTagList = None,
+    ) -> CreateDeliveryStreamOutput:
+        region = FirehoseBackend.get()
+
+        destinations: DestinationDescriptionList = []
+        if elasticsearch_destination_configuration:
+            destinations.append(
+                DestinationDescription(
+                    DestinationId=short_uid(),
+                    ElasticsearchDestinationDescription=convert_es_config_to_desc(
+                        elasticsearch_destination_configuration
+                    ),
+                )
+            )
+        if amazonopensearchservice_destination_configuration:
+            destinations.append(
```

Comment:
To me this implies that a Firehose can be configured to deliver to both an ElasticSearch and OpenSearch destination simultaneously. Is that accurate?

Reply:
Yes, that's correct. :)